### ISSUE TITLE
"Too many open files" error leads to dropping all peers, too many open connections?

### ISSUE THREAD

**Issue by drewstone** [paritytech/substrate#5243]
We see this error a non-negligible amount of times in Edgeware and perhaps it is due to there being too many open connections to peers? I had chalked it up to be a `ulimit` issue with respect to `systemd` settings but am hearing there are potentially other bugs (too many connections?)

**Comment by bkchr** [comment]
Which version of Substrate?

**Comment by drewstone** [comment]
https://github.com/paritytech/substrate/tree/00a400f82539e2f78e8ddbcd98aea512c87c5f3c

**Comment by bkchr** [comment]
Yeah that version is too old, could you please update? Meanwhile we have added some fixes to the networking that probably solves your problem.
